Maple2.Server.Game/PacketHandlers/InsigniaHandler.cs

```diff
@@ -0,0 +1,87 @@
+﻿using System;
+using Maple2.Database.Extensions;
+using Maple2.Database.Storage;
+using Maple2.Model.Enum;
+using Maple2.Model.Game;
+using Maple2.Model.Metadata;
+using Maple2.PacketLib.Tools;
+using Maple2.Server.Core.Constants;
+using Maple2.Server.Core.PacketHandlers;
+using Maple2.Server.Game.Packets;
+using Maple2.Server.Game.Session;
+
+namespace Maple2.Server.Game.PacketHandlers;
+
+public class InsigniaHandler : PacketHandler<GameSession> {
+    public override RecvOp OpCode => RecvOp.Insignia;
+
+    #region Autofac Autowired
+    // ReSharper disable MemberCanBePrivate.Global
+    public required TableMetadataStorage TableMetadata { private get; init; }
+    // ReSharper restore All
+    #endregion
+
+    public override void Handle(GameSession session, IByteReader packet) {
+        if (session.Field == null) {
+            return;
+        }
+
+        // Check and remove any existing insignia buffs
+        if (TableMetadata.InsigniaTable.Entries.TryGetValue(session.Player.Value.Character.Insignia, out InsigniaTable.Entry? oldSigniaMetadata) && oldSigniaMetadata.BuffId > 0) {
+            session.Player.Buffs.Remove(oldSigniaMetadata.BuffLevel);
+        }
+
+        short insigniaId = packet.ReadShort();
+        if (!TableMetadata.InsigniaTable.Entries.TryGetValue(insigniaId, out InsigniaTable.Entry? newSigniaMetadata)) {
+            return;
+        }
+
+        session.Player.Value.Character.Insignia = insigniaId;
+
+        bool display = false;
```

Comment:
looks like a good candidate for a switch expression:
```cs
bool display = newSigniaMetadata.Type switch {
    case InsigniaConditionType.title => session.Player.Value.Unlock.Titles.Contains(newSigniaMetadata.Code),
    case InsigniaConditionType.adventure_level => session.Player.Value.Unlock.Titles.Contains(newSigniaMetadata.Code),
    ...
```

Reply:
Nevermind, looks like there is a case where you need a statement and apparently they are not supposed by switch expressions. However you could simplify this a bit with:
```cs
case InsigniaConditionType.title:
    display = session.Player.Value.Unlock.Titles.Contains(newSigniaMetadata.Code);
    break;
```

Reply:
same can be done to most of the other cases as well